src/types/algorand-client-interface.ts

```diff
@@ -1,29 +1,5 @@
-import algosdk from 'algosdk'
-import { AlgorandClientTransactionCreator } from './algorand-client-transaction-creator'
-import { AlgorandClientTransactionSender } from './algorand-client-transaction-sender'
-import { AppDeployer } from './app-deployer'
-import { AppManager } from './app-manager'
-import { TransactionComposer } from './composer'
-import { NetworkDetails } from './network-client'
-import Algodv2 = algosdk.Algodv2
-import Indexer = algosdk.Indexer
+import { AlgorandClient } from './algorand-client'
+import { InterfaceOf } from './instance-of'
 
-/** Interface for the bulk of the `AlgorandClient` functionality.
- *
- * Used to take a dependency on AlgorandClient without generating a circular dependency.
- */
-export interface AlgorandClientInterface {
-  app: AppManager
-  appDeployer: AppDeployer
-  send: AlgorandClientTransactionSender
-  createTransaction: AlgorandClientTransactionCreator
-  newGroup(): TransactionComposer
-  client: {
-    algod: Algodv2
-    indexer?: Indexer
-    network(): Promise<NetworkDetails>
-    isLocalNet(): Promise<boolean>
-    isTestNet(): Promise<boolean>
-    isMainNet(): Promise<boolean>
-  }
-}
+/** AlgorandClient interface used to eleminate circular dependencies */
+export type AlgorandClientInterface = InterfaceOf<AlgorandClient>
```

Comment:
Why is this change in this PR?

Reply:
I had to either update the interface in this PR and deal with merge conflicts when #365 is merged or merge #365 in this branch. Although as per my latest comment on #365 I'm not sure if this is the best approach. fwiw if I encountered having to do this before #365, I probably would have made the same change.